renderer-macos/lib/bindings/BriskEffectView.re

```diff
@@ -1,90 +1,111 @@
-  let headerView = headerView();
-  external sheet: unit => t =
-    "ml_getNSVisualEffectMaterialSheet_bc" "ml_getNSVisualEffectMaterialSheet";
-  let sheet = sheet();
-  external windowBackground: unit => t =
-    "ml_getNSVisualEffectMaterialWindowBackground_bc"
-    "ml_getNSVisualEffectMaterialWindowBackground";
-  let windowBackground = windowBackground();
-  external hUDWindow: unit => t =
-    "ml_getNSVisualEffectMaterialHUDWindow_bc"
-    "ml_getNSVisualEffectMaterialHUDWindow";
-  let hUDWindow = hUDWindow();
-  external fullScreenUI: unit => t =
-    "ml_getNSVisualEffectMaterialFullScreenUI_bc"
-    "ml_getNSVisualEffectMaterialFullScreenUI";
-  let fullScreenUI = fullScreenUI();
-  external toolTip: unit => t =
-    "ml_getNSVisualEffectMaterialToolTip_bc"
-    "ml_getNSVisualEffectMaterialToolTip";
-  let toolTip = toolTip();
-  external contentBackground: unit => t =
-    "ml_getNSVisualEffectMaterialContentBackground_bc"
-    "ml_getNSVisualEffectMaterialContentBackground";
-  let contentBackground = contentBackground();
-  external underWindowBackground: unit => t =
-    "ml_getNSVisualEffectMaterialUnderWindowBackground_bc"
-    "ml_getNSVisualEffectMaterialUnderWindowBackground";
-  let underWindowBackground = underWindowBackground();
-  external pageBackground: unit => t =
-    "ml_getNSVisualEffectMaterialPageBackground_bc"
-    "ml_getNSVisualEffectMaterialPageBackground";
-  let pageBackground = pageBackground();
-  external appeareanceBased: unit => t =
-    "ml_getNSVisualEffectMaterialAppeareanceBased_bc"
-    "ml_getNSVisualEffectMaterialAppeareanceBased";
-  let appeareanceBased = appeareanceBased();
-  */
 };
 
 module BlendingMode = {
   type t;
+
   external behindWindow: unit => t =
     "ml_getNSVisualEffectBlendingModeBehindWindow_bc"
     "ml_getNSVisualEffectBlendingModeBehindWindow";
-  let behindWindow = behindWindow();
+
   external withinWindow: unit => t =
     "ml_getNSVisualEffectBlendingModeWithinWindow_bc"
     "ml_getNSVisualEffectBlendingModeWithinWindow";
-  let withinWindow = withinWindow();
 };
 
-type backgroundStyle =
-  | Raised
-  | Lowered;
+module EffectState = {
+  type t;
 
-type visualEffectState =
-  | FollowsWindowActiveState
-  | Active
-  | Inactive;
+  external followsWindowActiveState: unit => t =
+    "ml_getNSVisualEffectStateFollowsWindowActiveState_bc"
+    "ml_getNSVisualEffectStateFollowsWindowActiveState";
+
+  external active: unit => t =
+    "ml_getNSVisualEffectStateActive_bc" "ml_getNSVisualEffectStateActive";
+
+  external inactive: unit => t =
+    "ml_getNSVisualEffectStateInactive_bc" "ml_getNSVisualEffectStateInactive";
+};
+
+type material = [ | `Titlebar | `Selection | `Menu | `Popover | `Sidebar];
+type blendingMode = [ | `BehindWindow | `WithinWindow];
+type effectState = [ | `FollowsWindowActiveState | `Active | `Inactive];
+
+type style = [
+  | `Material(material)
+  | `BlendingMode(blendingMode)
+  | `Emphasized(bool)
+  | `EffectState(effectState)
```

Comment:
Nice! Does border radius work here?

Reply:
btw, shouldn't we have functions for that like we have for other styles?

Reply:
Hmmm, it won't. I dropped `View` style attributes.

Reply:
I wanted to add them, but they won't be available in the scope, not sure ``BriskEffectView.material(`Sidebar)`` would look better than `` `Material(`Sidebar)``

Reply:
I think we should be either using functions for everything or for nothing (I don't see how the middle ground helps, I see it more as a confusion). so EffectView.(material(sidebar)). Btw, I think we should re-export these inside the EffectView module for easy discovery.

Reply:
Updated, will that be enough?
Not sure I understand what exactly you want to re-export.